pallets/maintenance-mode/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// Edit this file to define custom logic or remove it if it is not needed.
+/// Learn more about FRAME and the core library of Substrate FRAME pallets:
+/// <https://docs.substrate.io/reference/frame-pallets/>
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod weights;
+pub use weights::*;
+
+pub const ACTIVATED: bool = true;
+pub const DEACTIVATEDD: bool = false;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use cumulus_primitives_core::{
+		relay_chain::BlockNumber as RelayBlockNumber, DmpMessageHandler,
+	};
+	use frame_support::{
+		pallet_prelude::{ValueQuery, *},
+		traits::Contains,
+	};
+	use frame_system::pallet_prelude::*;
+	pub use log;
+	use sp_std::vec::Vec;
+	use xcm_primitives::PauseXcmExecution;
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[derive(Default)]
+	#[pallet::genesis_config]
+	/// Genesis config for maintenance mode pallet
+	pub struct GenesisConfig {}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			MaintenanceModeStatus::<T>::put(ACTIVATED);
```

Comment:
so we can have the chance to check that everything is fine before making the operations available.

Reply:
I got it, but it doesn't seem to be a responsibility of a pallet. This decision (on or off at the start) should be taken on a composition root level. My suggestion here is to add `activated` into the genesis config
```
	#[pallet::genesis_config]
	/// Genesis config for lockdown mode pallet
	pub struct GenesisConfig<T: Config> {
		pub activated: bool,
		pub _phantom: PhantomData<T>,
	}

	impl<T: Config> Default for GenesisConfig<T> {
		fn default() -> Self {
			Self { activated: ACTIVATED, _phantom: PhantomData }
		}
	}

	#[pallet::genesis_build]
	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
		fn build(&self) {
			LockdownModeStatus::<T>::put(&self.activated);
		}
	}
```

this will also let you to set initial state for you tests
```
pub fn new_test_ext(activated: bool) -> sp_io::TestExternalities {
	let mut storage = system::GenesisConfig::default().build_storage::<Test>().unwrap();
	pallet_lockdown_mode::GenesisConfig::<Test> { activated, _phantom: PhantomData }
		.assimilate_storage(&mut storage)
		.unwrap();

	storage.into()
}
```

So that you don't need to activate lockdown mode first to deactivate it in the next step.

You can also add a test to make sure that the storage is initialized properly
```
#[test]
fn initialized() {
	[true, false].iter().for_each(|expected| {
		new_test_ext(*expected).execute_with(|| {
			let lockdown_mode = LockdownModeStatus::<Test>::get();
			assert_eq!(lockdown_mode, *expected);
		});
	});
}
```

Reply:
@hbulgarini I think @kalaninja makes sense. If you can apply those changes, we should be good then.

Reply:
https://github.com/paritytech/trappist/issues/172